app/policies/building_policy.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class BuildingPolicy < ApplicationPolicy
+  def index?
+    grower?
+  end
+
+  def show?
+    grower?
+  end
+
+  def create?
+    grower?
+  end
+
+  def update?
+    grower?
+  end
+
+  def destroy?
+    return false unless grower?
+
+    return true if record.greenhouses.flat_map(&:benches).flat_map(&:request_distributions).empty?
```

Comment:
utiliser des `flat_map` ici n'est pas optimal : ça force à charger toutes les associations SQL et à les mettre dans un tableau : non seulement ça génère des N+1, mais en plus ça fait travailler en ruby plutôt qu'en SQL, qui est plus lent.

Bon après je vois aussi que j'ai écrit ça dans d'autres policies à l'époque donc c'est normal que tu aies repris ça 😄 

Une meilleure approche serait de tout faire en SQL, comme je le disais, en utilisant par exemple des `through associations` => https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association

En ajoutant sur le modèle `Greenhouse` l'association suivante : `has_many :request_distributions, through: :benches`, tu pourras écrire directement `greenhouse.request_distributions` pour travailler avec l'association en SQL directement !

Et de même, en ajoutant sur ton modèle `Building` l'association `has_many :request_distributions, through: :greenhouses`, tu pourras utiliser ici `record.request_distributions.empty?`, ce qui sera quand même plus digeste à la lecture, et plus performant ! :)

Bon après ça n'est pas une priorité non plus hein, en soi ça marche comme ça et c'est déjà très bien ! Mais si on veut faire du code plus clean, ça serait pas mal de remplacer nos `map` / `flat_map` dans les policies par ça 😉